src/lib/cached/cached.mli

```diff
@@ -33,10 +33,19 @@ module Spec : sig
     -> autogen_path:string
     -> manual_install_path:string
     -> brew_install_path:string
+    -> s3_install_path:string
     -> digest_input:('input -> string)
     -> create_env:('input -> 'env)
     -> input:'input
     -> 'a t
 end
 
-val run : 'a Spec.t -> 'a Deferred.t
+(** A Semigroup for tracking the "dirty bit" of whether or not we've generated something or only received cache hits *)
+module Regenerated : sig
```

Comment:
I'm not sure this is the right name for this module. Maybe you don't regenerate anything and you load your keys for the first time. I think the code would be easier to understand if type t is `Did_manually_generate of bool` instead of a variant.

Reply:
I'll change the name, but I like the variant instead of a bool